transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -151,16 +156,9 @@ public final ChannelPipeline addFirst(String name, ChannelHandler handler) {
         return addFirst(null, name, handler);
     }
 
-    private enum AddStrategy {
```

Comment:
hmm.. I think the enum is actually better and more clear then static ints.

Reply:
- enum class consumes meta space
- takes time to load a class
- every enum entry is an actual object, that costs RAM
- switch over int is faster than switch over enum

As it is used only in 1 method and not publicly exposed, the enum usage is not justified.